pkg/pull/pull.go

```diff
@@ -236,7 +238,7 @@ func Pull(upstreamURI string, pullOptions PullOptions) (string, error) {
 			logger.Infof("Expecting to install version %s but airgap bundle version is %s.", fetchOptions.AppVersionLabel, airgap.Spec.VersionLabel)
 		}
 
-		if fetchOptions.License.Spec.ChannelID != airgap.Spec.ChannelID {
+		if _, err = kotsutil.FindChannelInLicense(airgap.Spec.ChannelID, fetchOptions.License); err != nil {
 			return "", util.ActionableError{
 				NoRetry: true, // if this is airgap upload, make sure to free up tmp space
 				Message: fmt.Sprintf("License (%s) and airgap bundle (%s) channels do not match.", fetchOptions.License.Spec.ChannelName, airgap.Spec.ChannelName),
```

Comment:
this error message will be confusing for multi-channel licenses as it's using the top-level channel name.

Reply:
fixed this up in this catch all commit: https://github.com/replicatedhq/kots/pull/4767/commits/f2624c8e7d5bb2a98151a79980d78de23806130f